Issues/193 fix all todos in html export

jpraetorius opened the pull request:
This not only addresses the Examples mentioned in #193 but additionaly

- exports Definitions
- adds Examples of fields in the Models section (code was there but never did anything)
- exports nested fields of fields in the Models section. This currently only goes one level deep, as it already is a proper pain without partials ;)

jpraetorius commented:
This is how things look like

![SCR-20240515-pbks](https://github.com/datacontract/datacontract-cli/assets/4026404/c822b42d-56e2-44d3-8ca7-6f6cd8a44d1a)

simonharrer commented:
jinja partials would be awesome

simonharrer commented:
layout of the flags definition seems broken

jpraetorius commented:
> layout of the flags definition seems broken

That was a simple showcase to test out how `enum` would render. It doesn't have any other properties defined that would go in the third column.
This is how it would be with some more content (but we can also try to move the enums to a different place)
![SCR-20240516-idpx](https://github.com/datacontract/datacontract-cli/assets/4026404/b3de3810-3dc2-4787-942a-b4faaa10b769)

jochenchrist commented:
I'd put them into the description column, as the enum list could become quite long.
To reduce abstraction, maybe I'd call them "Enum" instead of "Allowed values"?

jpraetorius commented:
So, here we go:
- introduced partials and theoretically fields now nest as deep as they are defined (I've only tried it two levels deep)
   - I'm not a 100% happy with how that is presented, some sort of collapse or different background shading would be good I think, but all of that is complicated to do with a table
- named enums enums and moved them into the description column. I've tried to make the labels stand out a bit more, so the contents of that column are a bit easier to parse
- moved all of the other blocks into partials, to keep the main template a bit more slim

jochenchrist commented:
The partials are a bit dangerous, as you need to remember to add them (as data files) to the MANIFEST.in
